src/components/popup/NewProductAdmin.js

```diff
@@ -74,42 +77,50 @@ export default class NewProductAdmin extends React.Component {
   onSubmit = (event) => {
     event.preventDefault();
     let formData = new FormData(event.target);
-
-    api.post('products', this.buildNewProductParams(formData))
+    console.log("data", this.buildNewProductParams());
+    api.post('products', this.buildNewProductParams())
       .then(response => {
-        this.uploadProductImage(response.data.product, formData)
+        // TODO this.uploadProductImage(response.data.product, formData)
+        console.log("response", response);
       })
       .catch(response => {
         console.log('error creating product ', response);
       });
   };
 
   uploadProductImage = (newProduct, formData) => {
-    imageApi.instance.headers.post['Content-Type'] = 'multipart/form-data';
-    imageApi.post(newProduct.image, formData.get("image"))
+    let config = {
+      headers: {
+        'Content-Type': 'multipart/form-data'
+      }
+    };
+    imageApi.post(newProduct.image, formData.get("image"), config)
       .then(response => {
         console.log('image uploaded ', response);
+        this.props.hideModals();
       })
       .catch(response => {
         console.log('error uploading image ', response);
       });
   };
 
-  buildNewProductParams = (formData) => {
-    let result = {};
-
-    for (let pair of formData.entries()) {
-      result[pair[0]] = pair[1];
-    }
-    result.category = "/api/categories/" + formData.get("category");
-    result.supplier = "/api/suppliers/" + formData.get("supplier");
-    result.subcategory = "/api/subcategories/" + formData.get("subcategory");
-    delete result.image;
-    return result;
+  buildNewProductParams = () => {
+    let data = {};
```

Comment:
přepiš to do podoby:
`
const data = {
  name: this.state.name,
  price: this. state.price,
...

`